### ISSUE TITLE
Detecting version from within magit as submodule does not work

### ISSUE THREAD

**Issue by arichiardi** [magit/magit#3648]
Hi!

>Also post the output of
>
 > M-x magit-version RET

This is the command I am trying to execute :smile: 

I am the following emacs situation: `magit` resides as submodule of my committed `.emacs.d` under the `$HOME/.emacs.d/packs/stable/git-pack/lib/magit` folder, which is a symlink to `$HOME/.emacs.d/packs/dev/git-pack/vendor/submodules/magit`. I inherited this situation from Emacs Live.

It looks like `magit-version` is trying to resolve the `magit` as repository:

```
not a git repository: $HOME/.emacs.d/packs/stable/git-pack/lib/magit/../../../../../../.git/modules/packs/dev/git-pack/vendor/submodules/magit
Cannot determine Magit’s version (error "$HOME/.emacs.d/packs/stable/git-pack/lib/magit/lisp/magit.el" repo t static elpa debug)
```

I know this is not very common and therefore I am not looking for a solution but a good workaround for getting the version. I see that sometimes `(magit-git-string "describe" "--tags" "--dirty")` works. But cannot reproduce the case when the symlink is followed like the above and things break.

I thank you in advance for your work on magit and any hint @tarsius :+1:

**Comment by arichiardi** [comment]
I tried to generate a `magit-version.el` but it looks like it gets deleted by:

```
(when (and static (not noninteractive))
  (ignore-errors (delete-file static)))
```

**Comment by tarsius** [comment]
Does the `$HOME` really appear in the output? If so, then that's the problem. Git doesn't expand variables (or `~`). Check the relevant `.git` and `.git/config` files for `$HOME`. If you cannot find any occurrence than you have to also check your emacs packages and configuration for occurrences.

**Comment by arichiardi** [comment]
No sorry that was just an example. `$HOME` is expanded to my folder which would be appearing above

**Comment by tarsius** [comment]
So just to be clear, the output actually was: 

```
not a git repository: /home/yourname/.emacs.d/packs/stable/git-pack/lib/magit/../../../../../../.git/modules/packs/dev/git-pack/vendor/submodules/magit
Cannot determine Magit’s version (error "/home/yourname/.emacs.d/packs/stable/git-pack/lib/magit/lisp/magit.el" repo t static elpa debug)
```

right?

I think the issue is that `/home/yourname/.emacs.d/packs/stable/git-pack/lib/magit/` contains

```
gitdir: ../../../../../../.git/modules/packs/dev/git-pack/vendor/submodules/magit
```

but should actually contain

```
gitdir: ../../../../../.git/modules/packs/dev/git-pack/vendor/submodules/magit
```

In any case, I don't think Magit is to blame here.

**Comment by arichiardi** [comment]
Yes it is like the above and you are right magit is not to blame, I was wondering how to workaround it :smile:

**Comment by tarsius** [comment]
As I said, I think the path in the file is wrong. If I am correct about that then correcting the path would fix the issue. If it is not that, then I don't know what it is.

By the way `magit-version` does work when submodules are involved -- that's how I have it installed myself.

**Comment by tarsius** [comment]
In the future please use one of the channels intended for support requests, as you were told when you opened this issue: https://github.com/magit/magit/issues/new?template=support_request.md. Thanks. That also makes it much more likely that someone else will come up with a more satisfying answer in the future.

**Comment by arichiardi** [comment]
@tarsius probably the detection is tricked by the symlink, as you see in the initial report of the issue. This is my unconfirmed hunch that I will try to confirm on my side.

And yes, I followed exactly what I have been told and opened a Reddit post [here](https://www.reddit.com/r/emacs/comments/9r1q7t/cannot_use_magitversion_in_custom_build/) before opening this one.

Then I debugged and saw that the detection mechanism is not working in my specific case. Thank you for attempting to explain the problem from a higher perspective. I will follow your suggestion and try to dig deeper.
